packages/nextjs/src/client-boundary/ClerkProvider.tsx

```diff
@@ -0,0 +1,19 @@
+'use client';
+
+import { useRouter } from 'next/compat/router';
+import React from 'react';
+
+import { ClientClerkProvider } from '../app-router/client/ClerkProvider';
+import { ClerkProvider as PageClerkProvider } from '../pages/ClerkProvider';
+import { type NextClerkProviderProps } from '../types';
+
+/**
+ * This is a compatibility layer to support a single ClerkProvider component in both the app and pages routers.
+ */
+export function ClerkProvider(props: NextClerkProviderProps) {
+  const router = useRouter();
+
+  const Provider = router ? PageClerkProvider : ClientClerkProvider;
```

Comment:
The solution here makes perfect sense to me :) 
Do you think we could merge the two client providers into one? I took a quick look and they look pretty similar.

Reply:
@nikosdouvlis possibly, the difficulty is they rely on the two different router hooks, which is why we need this layer. We could abstract the dependency on the router into a prop and use this layer to compute the prop to pass down to single provider.